### PR TITLE
Add step0 to paramscan

### PR DESCRIPTION
**Pull request by kavir1698**
Adds `step0` keywords argument to `paramscan`. Also adds an example to the tutorial showing how to aggregate the simulation results across replicates.

### CODE REVIEW

File: docs/src/tutorial.md [hunk 1]
```diff
@@ -302,3 +302,15 @@ parameters = Dict(:min_to_be_happy=>collect(2:5), :numagents=>[200,300], :griddi
 data = paramscan(parameters, initialize;
        properties=properties, n = 3, agent_step! = agent_step!)
 ```
+
+We can combine all replicates with an aggregating function, such as mean, using
+the `aggregate` function from the `DataFrames` package:
+
+```@example schelling
+import Statistics: mean
+data_mean = Agents.aggregate(data, [:step, :min_to_be_happy, :numagents],  mean)
```
Comment (Datseris) [inline docs/src/tutorial.md]: How much sense does it make to average over `:step` ? I mean scientifically, not programmatically?

**Comment by kavir1698** [comment]
> How much sense does it make to average over :step ? I mean scientifically, not programmatically?

When one records the output during the simulations, not only at the last step, he is interested in how the output changes, not just where it gets. Therefore, it is only reasonable to take the average of replicates per step.

**Comment by Datseris** [comment]
Okay, I think I have misunderstood what is happening here with `aggregate`. Can you please paste here the output of this new code block?

**Comment by kavir1698** [comment]
Oh yeah, I forgot to write `paramscan` with replicates. I will update the tutorial.

**Comment by Datseris** [comment]
@kavir1698 I think you have misunderstood how Documenter works. Please remove the copy/pasted dataframes from the examples. Documenter runs the **valid** julia code, and after the code it automatically attaches the output. Thus the created Dataframes will be automatically attached in the built documentation. 

I asked to copy/paste the result _here_ on this GitHub discussion, because I was confused about what the output should be.

**Comment by kavir1698** [comment]
This is the output of `paramscan` with 3 replicates:

```
96×5 DataFrames.DataFrame
│ Row │ happyperc(mood) │ step  │ replicate │ min_to_be_happy │ numagents │
│     │ Float64         │ Int64 │ Int64     │ Int64           │ Int64     │
├─────┼─────────────────┼───────┼───────────┼─────────────────┼───────────┤
│ 1   │ 0.0             │ 0     │ 1         │ 5               │ 300       │
│ 2   │ 0.586667        │ 1     │ 1         │ 5               │ 300       │
│ 3   │ 0.81            │ 2     │ 1         │ 5               │ 300       │
│ 4   │ 0.876667        │ 3     │ 1         │ 5               │ 300       │
│ 5   │ 0.0             │ 0     │ 2         │ 5               │ 300       │
│ 6   │ 0.61            │ 1     │ 2         │ 5               │ 300       │
│ 7   │ 0.853333        │ 2     │ 2         │ 5               │ 300       │
│ 8   │ 0.92            │ 3     │ 2         │ 5               │ 300       │
│ 9   │ 0.0             │ 0     │ 3         │ 5               │ 300       │
│ 10  │ 0.643333        │ 1     │ 3         │ 5               │ 300       │
⋮
│ 86  │ 0.365           │ 1     │ 1         │ 2               │ 200       │
│ 87  │ 0.615           │ 2     │ 1         │ 2               │ 200       │
│ 88  │ 0.75            │ 3     │ 1         │ 2               │ 200       │
│ 89  │ 0.0             │ 0     │ 2         │ 2               │ 200       │
│ 90  │ 0.385           │ 1     │ 2         │ 2               │ 200       │
│ 91  │ 0.555           │ 2     │ 2         │ 2               │ 200       │
│ 92  │ 0.69            │ 3     │ 2         │ 2               │ 200       │
│ 93  │ 0.0             │ 0     │ 3         │ 2               │ 200       │
│ 94  │ 0.38            │ 1     │ 3         │ 2               │ 200       │
│ 95  │ 0.575           │ 2     │ 3         │ 2               │ 200       │
│ 96  │ 0.71            │ 3     │ 3         │ 2               │ 200       │
```

And this is the output of `aggregate`:

```
32×5 DataFrames.DataFrame
│ Row │ step  │ min_to_be_happy │ numagents │ happyperc(mood)_mean │ replicate_mean │
│     │ Int64 │ Int64           │ Int64     │ Float64              │ Float64        │
├─────┼───────┼─────────────────┼───────────┼──────────────────────┼────────────────┤
│ 1   │ 0     │ 5               │ 300       │ 0.0                  │ 2.0            │
│ 2   │ 1     │ 5               │ 300       │ 0.613333             │ 2.0            │
│ 3   │ 2     │ 5               │ 300       │ 0.833333             │ 2.0            │
│ 4   │ 3     │ 5               │ 300       │ 0.908889             │ 2.0            │
│ 5   │ 0     │ 4               │ 300       │ 0.0                  │ 2.0            │
│ 6   │ 1     │ 4               │ 300       │ 0.592222             │ 2.0            │
│ 7   │ 2     │ 4               │ 300       │ 0.835556             │ 2.0            │
│ 8   │ 3     │ 4               │ 300       │ 0.908889             │ 2.0            │
│ 9   │ 0     │ 3               │ 300       │ 0.0                  │ 2.0            │
│ 10  │ 1     │ 3               │ 300       │ 0.625556             │ 2.0            │
│ 11  │ 2     │ 3               │ 300       │ 0.821111             │ 2.0            │
│ 12  │ 3     │ 3               │ 300       │ 0.904444             │ 2.0            │
│ 13  │ 0     │ 2               │ 300       │ 0.0                  │ 2.0            │
│ 14  │ 1     │ 2               │ 300       │ 0.604444             │ 2.0            │
│ 15  │ 2     │ 2               │ 300       │ 0.822222             │ 2.0            │
│ 16  │ 3     │ 2               │ 300       │ 0.906667             │ 2.0            │
│ 17  │ 0     │ 5               │ 200       │ 0.0                  │ 2.0            │
│ 18  │ 1     │ 5               │ 200       │ 0.421667             │ 2.0            │
│ 19  │ 2     │ 5               │ 200       │ 0.591667             │ 2.0            │
│ 20  │ 3     │ 5               │ 200       │ 0.718333             │ 2.0            │
│ 21  │ 0     │ 4               │ 200       │ 0.0                  │ 2.0            │
│ 22  │ 1     │ 4               │ 200       │ 0.33                 │ 2.0            │
│ 23  │ 2     │ 4               │ 200       │ 0.591667             │ 2.0            │
│ 24  │ 3     │ 4               │ 200       │ 0.718333             │ 2.0            │
│ 25  │ 0     │ 3               │ 200       │ 0.0                  │ 2.0            │
│ 26  │ 1     │ 3               │ 200       │ 0.33                 │ 2.0            │
│ 27  │ 2     │ 3               │ 200       │ 0.608333             │ 2.0            │
│ 28  │ 3     │ 3               │ 200       │ 0.728333             │ 2.0            │
│ 29  │ 0     │ 2               │ 200       │ 0.0                  │ 2.0            │
│ 30  │ 1     │ 2               │ 200       │ 0.376667             │ 2.0            │
│ 31  │ 2     │ 2               │ 200       │ 0.581667             │ 2.0            │
│ 32  │ 3     │ 2               │ 200       │ 0.716667             │ 2.0            │
```

**Comment by Datseris** [comment]
Aaaaah I see. Yeap this makes a lot of sense. So in principle you are just averaging the happy percentages across replicates, right? Cool, and super useful!!!

As a comment though: the column `replicate_mean` is kind of useless. Maybe drop it from being displayed in the documentation (by just removing this column from the dataframe, using `delete!`).

File: docs/src/tutorial.md [hunk 2]
```diff
@@ -302,3 +302,22 @@ parameters = Dict(:min_to_be_happy=>collect(2:5), :numagents=>[200,300], :griddi
...
+```
+
+Note that the second argument takes the column names on which to split the data. It should include
```
Comment (Datseris) [inline docs/src/tutorial.md]: I think we can make this just a bit more clear:

> Note that the second argument takes the column names on which to split the data. I.e. it denotes which columns should _not_ be aggregated.

**Comment by kavir1698** [comment]
> So in principle you are just averaging the happy percentages across replicates, right?

Exactly!
The new commit addresses your suggestions.

**Pull request merged**